From 9e4b31c0ba14c3e5613dec51aa0d086fe3ae28ea Mon Sep 17 00:00:00 2001
From: Yaroslav Halchenko <debian@onerussian.com>
Date: Tue, 28 Sep 2021 11:21:14 -0400
Subject: [PATCH 2/2] BF: ProgressHandler - close the other handler if was
 specified

This makes
DATALAD_ASSERT_NO_OPEN_FILES=assert python -m nose -s -v datalad/tests/test_log.py:test_logging_to_a_file
to pass
---
 datalad/log.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/datalad/log.py b/datalad/log.py
index 9e31c7a98..4cab62ba3 100644
--- a/datalad/log.py
+++ b/datalad/log.py
@@ -210,6 +210,11 @@ class ProgressHandler(logging.Handler):
         self._other_handler = other_handler
         self.pbars = {}
 
+    def close(self):
+        if self._other_handler:
+            self._other_handler.close()
+        super().close()
+
     def emit(self, record):
         from datalad.ui import ui
         if not hasattr(record, 'dlm_progress'):
-- 
2.33.0

